src/EnergyPlus/MixedAir.cc

```diff
@@ -3279,6 +3279,155 @@ namespace MixedAir {
 						SetupEMSActuator( "Outdoor Air Controller", OAController( OAControllerLoop ).Name, "Air Mass Flow Rate", "[kg/s]", OAController( OAControllerLoop ).EMSOverrideOARate, OAController( OAControllerLoop ).EMSOARateValue );
 					}
 
+					VentMechObjectNum = OAController( OAControllerLoop ).VentMechObjectNum;
+					if ( VentMechObjectNum > 0 ){
+						if (!VentilationMechanical( VentMechObjectNum ).DCVFlag){
+							AirLoopControlInfo( thisAirLoop ).AirLoopDCVFlag = false;
+						}
+
+						// moved code below here
+
+						if ( MechVentCheckFlag( OAControllerLoop ) ) {
+							// Make these checks only once at the beginning of the simulation
+
+							// Make sure all air loop zones and air loop zones with people objects are covered by mechanical ventilation
+							// Issue a warning only if the zone is not accounted for in the associated mechanical ventilation object
+							auto & vent_mech(VentilationMechanical(VentMechObjectNum));
+
+							// Make sure all zones with mechanical ventilation are on the correct air loop
+							TempMechVentArrayCounter = 0;
+							for (NumMechVentZone = 1; NumMechVentZone <= vent_mech.NumofVentMechZones; ++NumMechVentZone) {
+								ZoneNum = vent_mech.Zone(NumMechVentZone);
+								auto const & zone(Zone(ZoneNum));
+								FoundZone = false;
+								for (AirLoopZoneInfoZoneNum = 1; AirLoopZoneInfoZoneNum <= AirLoopZoneInfo(AirLoopNum).NumZones; ++AirLoopZoneInfoZoneNum) {
```

Comment:
Getting closer, but this still crashes with the #5476 defect file.  In this line here AirLoopNum should be thisAirLoop also in the next line, and anywhere else within this block.